File: Programming-Basics-with-Python/9.While Loop - Lab/08. Graduation.py
student_name = input()
current_year_grade = 0
grades_counter = 0
class_year = 1
average_grade = 0
failed_year = 0

while class_year <= 12:
    current_year_grade = float(input())
    if current_year_grade >= 4:
        class_year += 1
        grades_counter += current_year_grade
    else:
        failed_year += 1
        if failed_year > 1:
            break

average_grade = grades_counter / 12
if class_year == 13:
    print(f"{student_name} graduated. Average grade: {average_grade:.2f}")

else:
    print(f"{student_name} has been excluded at {class_year} grade")

File: Programming-Basics-with-JavaScript/5.While Loop/08graduation.js
function graduation(input){
    let idx = 0
    let i = 1
    let correction = 0
    let boolTrue = true
    let sumGrades = 0
    let name = input[idx]
    idx++
    let grade = input[idx]
    idx++
    while (i <= 12){
        grade = Number(grade)
        if (grade < 4){
            correction += 1
            if(correction < 1){
                continue
            }else {
                boolTrue = false
                break
            }
            
        }
        sumGrades += grade
        grade = input[idx]
        idx++
        i++
    }
    
    if (boolTrue){
        console.log(`${name} graduated. Average grade: ${(sumGrades / 12).toFixed(2)}`)
    }else {
        console.log(`${name} has been excluded at ${i} grade`)
    }

}
graduation(["Gosho", 
"5",
"5.5",
"6",
"5.43",
"5.5",
"6",
"5.55",
"5",
"6",
"6",
"5.43",
"5"])


